src/test/java/api/loans/OverrideCheckOutByBarcodeTests.java

```diff
@@ -0,0 +1,297 @@
+
+    assertThat(loan.getString("id"), is(notNullValue()));
+
+    assertThat("user ID should match barcode",
+      loan.getString("userId"), is(steve.getId()));
+
+    assertThat("item ID should match barcode",
+      loan.getString("itemId"), is(smallAngryPlanet.getId()));
+
+    assertThat("status should be open",
+      loan.getJsonObject("status").getString("name"), is("Open"));
+
+    assertThat("action should be checkedout",
+      loan.getString("action"), is(CHECKED_OUT_THROUGH_OVERRIDE));
+
+    assertThat("loan date should be as supplied",
+      loan.getString("loanDate"), isEquivalentTo(TEST_LOAN_DATE));
+
+    assertThat("due date should be 3 weeks after loan date, based upon loan policy",
+      loan.getString("dueDate"), isEquivalentTo(TEST_DUE_DATE));
+
+    assertThat("Checkout service point should be stored",
+      loan.getString("checkoutServicePointId"), is(checkoutServicePointId));
+
+    smallAngryPlanet = itemsClient.get(smallAngryPlanet);
+
+    assertThat(smallAngryPlanet, hasItemStatus(CHECKED_OUT));
+
+    assertThat("has item information",
+      loan.containsKey("item"), is(true));
+
+    assertThat("title is taken from instance",
+      loan.getJsonObject("item").getString("title"),
+      is("The Long Way to a Small, Angry Planet"));
+
+    assertThat("barcode is taken from item",
+      loan.getJsonObject("item").getString("barcode"),
+      is("036000291452"));
+
+    assertThat("call number is 123456", loan.getJsonObject("item")
+      .getString("callNumber"), is("123456"));
+
+    assertThat(loan.getJsonObject("item").encode() + " contains 'materialType'",
+      loan.getJsonObject("item").containsKey("materialType"), is(true));
+
+    assertThat("materialType is book", loan.getJsonObject("item")
+      .getJsonObject("materialType").getString("name"), is("Book"));
+
+    assertThat("item has contributors",
+      loan.getJsonObject("item").containsKey("contributors"), is(true));
+
+    JsonArray contributors = loan.getJsonObject("item").getJsonArray("contributors");
+
+    assertThat("item has a single contributor",
+      contributors.size(), is(1));
+
+    assertThat("Becky Chambers is a contributor",
+      contributors.getJsonObject(0).getString("name"), is("Chambers, Becky"));
+
+    assertThat("has item status",
+      loan.getJsonObject("item").containsKey("status"), is(true));
+
+    assertThat("status is taken from item",
+      loan.getJsonObject("item").getJsonObject("status").getString("name"),
+      is("Checked out"));
+
+    assertThat("Should not have snapshot of item status, as current status is included",
+      loan.containsKey("itemStatus"), is(false));
+
+    assertThat("item has location",
+      loan.getJsonObject("item").containsKey("location"), is(true));
+
+    assertThat("location is taken from holding",
+      loan.getJsonObject("item").getJsonObject("location").getString("name"),
+      is("3rd Floor"));
+  }
+
+  @Test
+  public void cannotOverrideCheckoutWhenItemIsLoanable()
```

Comment:
There is an implicit assumption in this test that the default loan policy is loanable. Whilst this is likely, I think it might be more explicit to set this up specifically during the test

Reply:
Done